fetch/mssqlserver/ercoleAgentMsSQLServer-Gather.ps1

```diff
@@ -0,0 +1,273 @@
+﻿# Copyright (c) 2020 Sorint.lab S.p.A.
+# 
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+# 
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+# 
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+param(
+    [Parameter(Mandatory=$false)][string]$instance ="DESKTOP-QN4D1T4\SQL2014",
+    [Parameter(Mandatory=$false)][string]$sqlDir ="..\..\sql\mssqlserver\",
+    [Parameter(Mandatory=$false)][string]$outDir ="..\..\Output\",
+    [Parameter(Mandatory=$false)][ValidateSet("json")] [string]$outputType ="json",
+    [Parameter(Mandatory=$false)][ValidateSet("file","object")] [string]$outputAs ="object",
+    [Parameter(Mandatory=$false)][ValidateSet("all","dbmounted", "edition", "listDatabases","db", "backup_schedule", "dbstatus", "schema", "ts", "segment_advisor","patches","psu-1","sqlFeatures")][string]$action = "all"
+)
+
+function executeQueue([System.Collections.ArrayList]$queryList, [System.Collections.ArrayList]$dbQueryList){
+    $resultList = new-object System.Collections.ArrayList;
+    $queryList |ForEach-Object {
+        $itemList = $_;
+        Write-Output $($($itemList.order.ToString()) + ':'+ $($itemList.queryPath) +':'+ $($itemList.outputFile)) |Out-File -FilePath $("$outDir"+'gather.log') -Append
```

Comment:
It's necessary to write files? Can you instead save the stdout in a variable or at least write to a temporary file?